content/menu.md

```diff
@@ -409,8 +409,8 @@ site_menu:
     visible: false
   - category: Streamlit library / Advanced features/ Add statefulness to apps
     url: /library/advanced-features/session-state
-  - category: Streamlit library / Advanced features/ Widget semantics
-    url: /library/advanced-features/widget-semantics
+  - category: Streamlit library / Advanced features/ Widget behavior
+    url: /library/advanced-features/widget-behavior
```

Comment:
This new page is amazing, and really thorough. I love the level of detail and clear explanations.

It could be a later PR, but I think we'll want to _at least_ link to this page from [here](https://docs.streamlit.io/library/advanced-features/session-state) and [here](https://docs.streamlit.io//library/api-reference/session-state) since they cover some overlapping content in less detail. We might want to do some rewrites on those pages as well (such as removing fully duplicated content and just having a brief summary and pointer to relevant part of this page).

Reply:
I think this would be a nice fast-follow so I can read those pages and make some light edits to reduce duplication as you say. I started this article on my personal account before I was hired, so it will be good to close out that "previous" work and start a new, internal PR. I do have "Rewrite session state docs" on my backlog.

Reply:
Hmmm. For my edification, what would be the nitty-gritty, technically correct description of what Streamlit is doing here to match a function to existing widgets @AnOctopus?